powershell-adapter/powershell.resource.ps1

```diff
@@ -1,427 +1,349 @@
-        elseif ($r.ParentPath) { $moduleName = Split-Path $r.ParentPath | Split-Path | Split-Path -Leaf }
-
-        $propertyList = @()
-        foreach ($p in $r.Properties)
-        {
-            if ($p.Name)
-            {
-                $propertyList += $p.Name
-            }
+        # workaround: if the resource does not have a module name, get it from parent path
+        # workaround: modulename is not settable, so clone the object without being read-only
+        $DscResourceInfo = [dscResource]::new()
+        $dsc.PSObject.Properties | ForEach-Object -Process { $DscResourceInfo.$($_.Name) = $_.Value }
+        if ($dsc.ModuleName) {
+            $moduleName = $dsc.ModuleName
         }
-
-        $fullResourceTypeName = "$moduleName/$($r.ResourceType)"
-        $script:ResourceCache[$fullResourceTypeName] = $r
-        if ($WinPS) {$requiresString = "Microsoft.Windows/WindowsPowerShell"} else {$requiresString = "Microsoft.DSC/PowerShell"}
-
-        $t = [Type]$r.ResourceType
-        $exportMethod = $t.GetMethod('Export')
-
-        $capabilities = @('Get', 'Set', 'Test')
-        if ($null -ne $exportMethod) {
-            $capabilities += 'Export'
+        elseif ($dsc.ParentPath) {
+            $moduleName = Split-Path $dsc.ParentPath | Split-Path | Split-Path -Leaf
+            $DscResourceInfo.Module = $moduleName
+            $DscResourceInfo.ModuleName = $moduleName
+            # workaround: populate module version from psmoduleinfo if available
+            if ($moduleInfo = Get-Module -Name $moduleName -ListAvailable -ErrorAction Ignore) {
+                $moduleInfo = $moduleInfo | Sort-Object -Property Version -Descending | Select-Object -First 1
+                $DscResourceInfo.Version = $moduleInfo.Version.ToString()
+            }
         }
 
-        $z = [pscustomobject]@{
-            type = $fullResourceTypeName;
-            kind = 'Resource';
-            version = $version_string;
-            capabilities = $capabilities;
-            path = $r.Path;
-            directory = $r.ParentPath;
-            implementedAs = $r.ImplementationDetail;
-            author = $author_string;
-            properties = $propertyList;
-            requires = $requiresString
+        $resourceCache += [resourceCache]@{
+            Type            = "$moduleName/$($dsc.Name)"
+            DscResourceInfo = $DscResourceInfo
         }
-
-        $z | ConvertTo-Json -Compress
     }
+    return $resourceCache
 }
-elseif ($Operation -eq 'Get')
-{
-    $result = @()
-
-    RefreshCache
+$resourceCache = Invoke-CacheRefresh
+
+# Convert the INPUT to a configFormat object so configuration and resource are standardized as moch as possible
+function Get-ConfigObject {
+    param(
+        [Parameter(Mandatory = $true, ValueFromPipeline = $true)]
+        $stdinput
+    )
+    # normalize the INPUT object to an array of configFormat objects
+    $inputObj = $stdInput | ConvertFrom-Json -Depth 10
```

Comment:
Since we are converting FROM json and not TO json, we shouldn't set explicit depth as we may be losing important parts of the input

```suggestion
    $inputObj = $stdInput | ConvertFrom-Json
```

Reply:
fixed